src/io/WKBReader.cpp

```diff
@@ -237,29 +237,29 @@ WKBReader::readGeometry()
         ordValues.resize(inputDimension);
     }
 
-    Geometry* result;
+    std::unique_ptr<Geometry> result;
 
     switch(geometryType) {
     case WKBConstants::wkbPoint :
-        result = readPoint();
+        result.reset(readPoint().release());
```

Comment:
This looks unusual - any reason why std::move couldn't work here instead?

Reply:
Hmm, I didn't think C++11 let me convert from a subclass `unique_ptr` to a base class `unique_ptr`. But I just tried it locally, and it seems to work. Explicit move isn't necessary since `readPoint()` is already an rvalue. Thanks for the catch!